### DISCUSSION TITLE
How to correct using 'Blocked' page functionally?

### DISCUSSION THREAD

**Discussion by m0zgen** [0xERR0R/blocky#391] [Q&A] [answered]
Hello, today I tried different methods to forward users to 'blocked' page from restricted domains. For this task I'm using `blockType: <custom IP>` configuration parameter, clients correctly receive response from `blocky` server defined <custom IP>, but if client trying to request to restricted domain located on https, he is receive expected message in browser `SSL_ERROR_BAD_CERT_DOMAIN`...

Maybe have opportunity in future add functionally to `blocky` to display to users html page or plain text response from blocky server directly for blocked / restricted domains?

**Comment by 0xERR0R** [comment, marked as answer]
Basically, you need a web server which serves on ports 80 and 443. This server should return custom html page on each root and *.html request. This server can also return some custom pictures or only single pixel for *.jpeg/png etc requests.

This would work well for HTTP requests. For HTTPS you need to make some kind of "man-in-the-middle attack". Your web server must generate certificates for hostnames in request on-the-fly. To avoid warnings on client side, you must import the root certificate of your server.

To summarize, since almost all traffic is encrypted today, it is probably to much effort to use the custom block page

**Reply by m0zgen** [reply]
oh yes,  it is really probably to much effort to use the custom block page, thanks a lot for the answer)
